libsolidity/ast/Types.h

```diff
@@ -54,6 +54,19 @@ using rational = boost::rational<bigint>;
 using TypeResult = util::Result<TypePointer>;
 using BoolResult = util::Result<bool>;
 
+}
+
+namespace solidity::langutil
+{
+
+bigint storageSizeUpperBound(frontend::Type const& _type);
+std::vector<frontend::Type const*> oversizedSubtypes(frontend::Type const& _type);
+
+}
```

Comment:
I'm wondering, if there's a better place for these than them being free functions in Types.h...

Reply:
Could be static (or virtual) functions within `Type`. That can be done as a part of refactoring `validForLocation`, `storageSizeUpper` and `oversizedSubtypes`.

Reply:
I'd be fine with leaving it here for now... but why put it in the other namespace then?

Reply:
I was not sure where the functions should live :). Moved to `solidity::frontend`. (But I plan to refactor them, together with `validForLocation`, anyway.)